src/pages/StocktakeEditPage.js

```diff
@@ -159,15 +158,18 @@ export class StocktakeEditPage extends React.Component {
    */
   onEndEditing = (key, stocktakeItem, newValue) => {
     const { database } = this.props;
-    const { reasons } = this.state;
+    const { reasons, isReasonsModalOpen } = this.state;
 
     if (key !== 'countedTotalQuantity' || newValue === '') return;
     const quantity = parsePositiveInteger(newValue);
     if (quantity === null) return;
+    // If the reason modal is open just ignore any change to the current line
+    // This a hack to solve https://github.com/openmsupply/mobile/issues/1011
+    // Underlying issue requires data table rewrite
+    if (isReasonsModalOpen) return;
```

Comment:
I think logic is clearer if this is at the top of the method straight after deconstructing. No need to parse anything if the data is wrong.